server/item/stack.go

```diff
@@ -326,6 +337,12 @@ func (s Stack) Comparable(s2 Stack) bool {
 	return true
 }
 
+// Variants returns true if multiple variants are allowed for the item stack. This is used mainly for crafting recipe
+// inputs.
+func (s Stack) Variants() bool {
```

Comment:
I don't really know about adding this and `WithVariants`. This is purely for recipes and the method doesn't actually mean anything to the Stack.

Reply:
I wouldn't say this necessarily - it can be useful externally, too, for example, using `Comparable` for only the type of item. It also prevents the entire "RecipeItem" wrapper too.